scripts/package

```diff
@@ -49,20 +40,32 @@ cd ../bin
 
 if [ ! -d build/system-charts ]; then
     mkdir -p build
-    git clone --branch $SYSTEM_CHART_DEFAULT_BRANCH https://github.com/rancher/system-charts build/system-charts
+    git clone --branch $SYSTEM_CHART_DEFAULT_BRANCH https://github.com/rancher/system-charts $SYSTEM_CHART_REPO_DIR
 fi
 
 if [ ! -d build/charts ]; then
-    git clone --branch $CHART_DEFAULT_BRANCH https://github.com/rancher/charts build/charts
-
-    # Iterate through chart directories and execute callback to extract latest version tgz
-    find build/charts/assets -type d -maxdepth 1 -exec bash -c 'extract_latest_version_tgz {}' \;
+    INDEX_PATH=$CHART_REPO_DIR/index.yaml
+    git clone --branch $CHART_DEFAULT_BRANCH https://github.com/rancher/charts $CHART_REPO_DIR
+
+    # Get list of paths to tarballs of the latest version of each chart from Helm repo's index.yaml
+    LATEST_TGZ_PATHS=$(yq r $INDEX_PATH "entries.*.[0].urls[0]")
```

Comment:
I suppose urls should never be blank, but if it is would yq error out here ? Same for entries actually, I wonder how a blank index.yaml reacts, you can make one with `helm repo index .`

Reply:
For blank index.yaml (completely blank file), entries or urls, the result would be that nothing or that specific url will not get extracted and images will not get included in the images list because it doesn't exist or is invalid (missing url). yq will not throw an error.

Reply:
Ok, I just want to ensure it doesn't error and break build. Same would go for `urls[0]` ? Should be easy enough to manually edit the index file and check by hand.

Reply:
I tested it with a blank index.yaml, empty entries as in `entries: {}` which is what `helm repo index .` creates, and a chart without a `urls` key. There were no error from yq. In the missing/empty entries case nothing was extracted. In the missing urls case, only that specific chart didn't get extracted.